include/El/core/DistGraph.hpp

```diff
@@ -11,6 +11,206 @@
 #ifndef EL_CORE_DISTGRAPH_HPP
 #define EL_CORE_DISTGRAPH_HPP
 
-#include <El/core/DistGraph/decl.hpp>
```

Comment:
Hi Jed,

Not sure this is accurate. This is a pure header only implementation, with
separate sources, so you have to spend more time because its not all in one
place. However, I agree that there is value for only having signatures. If
impl.hpp was impl.cpp, then what you are saying is accurate.

Right now the headers extern template instantiations for a zillion things,
this is assuredly makes the builds *very* slow, because template
_instantiation_ is relatively slow.

I'm trying to back out these changes for now.

Best,
-rhl

On Mon, Aug 22, 2016 at 4:19 PM Jed Brown <notifications@github.com> wrote:

> In include/El/core/DistGraph.hpp
> <https://github.com/elemental/Elemental/pull/132#discussion_r75777403>:
>
> > @@ -11,6 +11,206 @@
> >  #ifndef EL_CORE_DISTGRAPH_HPP
> >  #define EL_CORE_DISTGRAPH_HPP
> >
> > -#include <El/core/DistGraph/decl.hpp>
>
> @rhl- <https://github.com/rhl-> In general, you're asking to spend way
> more time compiling if you combine them. It is very common for header
> dependencies and transitive dependencies (header or source) to need only
> the declarations, which means that all those compilation units don't need
> to pull in the implementation and don't need to be recompiled any time the
> impl.hpp changes (usually much more common than decl.hpp changes). We saw a
> tangible improvement in PETSc compile times (especially incremental) when
> we split some headers.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/elemental/Elemental/pull/132/files/32571ce8e750f142519b75f791742417c39b4bf9#r75777403>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATdUWJiGKjYBPYjywtwnQin6eGRvX_Oks5qii5lgaJpZM4HVOQ_>
> .
>